test/tools/yulInterpreter/Interpreter.cpp

```diff
@@ -145,6 +145,11 @@ void Interpreter::operator()(ForLoop const& _forLoop)
 	}
 	while (evaluate(*_forLoop.condition) != 0)
 	{
+		// Increment step for each loop iteration for loops with
```

Comment:
Could we also increment the step when entering a block?

Reply:
Actually, that is what we did before. However that introduced a different problem (see https://github.com/ethereum/solidity/issues/6476). To summarize: If we increment step when entering a block, then the optimized code needs a higher threshold because it may introduce additional blocks.

Reply:
To be honest, the solution in this PR (counting statements instead of blocks) does not solve the underlying problem of optimised code needing a different threshold. This is because optimiser step move loop condition into body creates at least one additional statement per iteration, thereby inducing more increments for optimised code.

This PR "solves" this by bailing out early if unoptimised code has a large trace: https://github.com/ethereum/solidity/pull/9459/files#diff-fc66c76b16410009767757a886824a1eR102